src/local-run.ts

```diff
@@ -1,51 +1,131 @@
-import { createManifest, readAll } from "./deps.ts";
-import { ParsePayload } from "./parse-payload.ts";
-import { DispatchPayload } from "./dispatch-payload.ts";
+import { createManifest, parse } from "./deps.ts";
+
+const SLACK_DEV_DOMAIN_FLAG = "sdk-slack-dev-domain";
+
+/**
+ * @param args An array of command line flags
+ * @returns The value of the SLACK_DEV_DOMAIN_FLAG flag, or empty string
+ */
+const parseDevDomain = (args: string[]): string => {
+  const flags = parse(args);
+  return flags[SLACK_DEV_DOMAIN_FLAG] ?? "";
+};
+
+/**
+ * Returns a module URL string pointing to `file` in the same
+ * directory as `mainModule`.
+ * @param mainModule The currently running Deno module, like from `Deno.mainModule`
+ * @param filename The name of the file to return the URL for
+ * @returns
+ */
+const findRelativeFile = (mainModule: string, filename: string): string => {
+  const moduleUrl = new URL(mainModule);
+  const path = moduleUrl.pathname;
+  const pathComponents = path.split("/");
+  if (pathComponents.length > 0) {
+    pathComponents[pathComponents.length - 1] = filename;
+  } else {
+    pathComponents[0] = filename;
+  }
+  moduleUrl.pathname = pathComponents.join("/");
+  return moduleUrl.href;
+};
+
+/**
+ * Determines the command line for the `deno run` invocation that will actually run the function,
+ * setting the appropriate permissions flags
+ * @param mainModule The URL to the main file being run, from `Deno.mainModule`
+ * @param denoExecutablePath The path to the deno executable
+ * @param manifest The application's manifest
+ * @param devDomain The domain of the slack dev instance being used, or empty string for production
+ * @returns The commandline to run `local-run-function.ts` to actually execute the function
+ */
+export const getCommandline = function (
+  mainModule: string,
+  denoExecutablePath: string,
+  // deno-lint-ignore no-explicit-any
+  manifest: any,
+  devDomain: string,
+): string[] {
+  const command = [
+    denoExecutablePath,
+    "run",
+    "-q",
+    "--config=deno.jsonc",
+    "--allow-read",
+  ];
+
+  const allowedDomains = manifest.outgoing_domains ?? [];
+
+  // If using a dev instance, allow making API calls to that domain
+  // and ignore SSL errors to it
+  if (devDomain !== "") {
+    command.push("--unsafely-ignore-certificate-errors=" + devDomain);
+    allowedDomains.push(devDomain);
+  } else {
+    allowedDomains.push("slack.com");
```

Comment:
the code path here is a slack internal dev using a numbered dev instance. in that case, api calls to slack.com wont work, since youre working with a dev workspace with dev tokens.